controller_manager/src/controller_manager.cpp

```diff
@@ -3149,6 +3300,27 @@ void ControllerManager::controller_manager_diagnostic_callback(
         "Resource Manager is not initialized properly!");
     }
   }
+
+  const double periodicity_error = std::abs(cm_stats.average - get_update_rate());
+  const std::string diag_summary =
+    "Controller Manager has bad periodicity : " + std::to_string(cm_stats.average) +
+    " Hz. Expected consistent " + std::to_string(get_update_rate()) + " Hz";
```

Comment:
could we please add a follow-up issue to make this statically allocated by using FMT?

Reply:
Done. Created! 
https://github.com/ros-controls/ros2_control/issues/1906